mqtt/mqtt-bridge/src/messages.rs

```diff
@@ -65,17 +65,23 @@ impl<S> StoreMqttEventHandler<S> {
                 // maps if local does not have a value it uses the topic that was received,
                 // else it checks that the received topic starts with local prefix and removes the local prefix
                 .map_or(Some(topic_name), |local_prefix| {
-                    let prefix = format!("{}/", local_prefix);
-                    topic_name.strip_prefix(&prefix)
+                    if local_prefix.is_empty() {
```

Comment:
just as a suggestion: can we make a new type `NonEmptyString` instead and pass it everywhere? with struct API similar to standard https://doc.rust-lang.org/std/num/struct.NonZeroUsize.html

Reply:
it this case you'll make sure it is not empty, and whenever you need a string you can extract it. as a convenience, you can make `fn as_str(&self) -> &str` and `impl AsRef<str> for NonEmptyString`